### PR TITLE
build: upgrade to node 14 lts

### PR DESCRIPTION
**Pull request by jamime**
### Proposed behaviour
Change GitHub CI so it uses a compatible version of node `>=14.16.0`(defined in package.json#engines).

I increased the version range to include `14.16.0`, this is latest version available for the cypress docker image. `14.16.1` was rather arbitrary, it was the latest `lts` version at the time the engine field was updated.

I've restricted the range to keep it on the current LTS (`14.x`).

We don't force our consumers to stay on `14`, but we don't officially support `15`.

We no longer need to transpile from `12` to `14` so I removed it from the `babelrc.json`.

I've also removed the other text editor recommendations. Developers are still free to choose their editor, but we can only provide support for VS Code.

### Current behaviour
GitHub CI is using `12.x`
### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
Blocking #3799

⚠️ Developers will no longer be able to start storybook if they have the wrong node version installed.

image

To upgrade developers can use `nvm install lts/fermium`

### Testing instructions
Ensure that GitHub checks pass.

### CODE REVIEW

**Review by sjohnson-sage — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

File: .github/workflows/cypress-push.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
             11, 12, 13, 14, 15,
             16, 17, 18, 19, 20
             ]
-    container: cypress/browsers:node12.18.3-chrome89-ff86
+    container: node14.16.0-chrome89-ff86
```
Comment (DlgSHi) [inline .github/workflows/cypress-push.yml:26]:
```suggestion
    container: cypress/browsers:node14.16.0-chrome89-ff86
```

File: .github/workflows/chromatic-pr.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
         fetch-depth: 0
     - uses: actions/setup-node@v1
       with:
-        node-version: '12'
+        node-version: '>=14.16.0 14'
```
Comment (DlgSHi) [inline .github/workflows/chromatic-pr.yml:18]: why we are passing `14.16.0` and `14` as a separate two values?

Comment (sjohnson-sage) [reply .github/workflows/chromatic-pr.yml:18]: `I've restricted the range to keep it on the current LTS (14.x).`

Comment (sjohnson-sage) [reply .github/workflows/chromatic-pr.yml:18]: It's saying node version 14.16.0 and above, but you can't go to v15 or above

File: .github/workflows/cypress-push.yml (hunk 1 above)
Comment (jamime) [reply .github/workflows/cypress-push.yml:26]: I've just updated, thanks.

File: .github/workflows/chromatic-pr.yml (hunk 2 above)
Comment (jamime) [reply .github/workflows/chromatic-pr.yml:18]: This means any version of 14 greater than 14.16.0 but not 15.

Comment (DlgSHi) [reply .github/workflows/chromatic-pr.yml:18]: Understood :)

**Review by sjohnson-sage — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**